cl/lib/elasticsearch_utils.py

```diff
@@ -3014,3 +3024,20 @@ def do_es_alert_estimation_query(
     estimation_query, _ = build_es_base_query(search_query, cd)
 
     return estimation_query.count()
+
+
+def simplify_estimated_count(search_count: int) -> int:
+    """Simplify the estimated search count to the nearest rounded figure.
+    It only applies this rounding if the search_count exceeds the
+    ELASTICSEARCH_CARDINALITY_PRECISION threshold.
+
+    :param search_count: The original search count.
+    :return: The simplified search_count, rounded to the nearest significant
+    figure or the original search_count if below the threshold.
+    """
+    if search_count > settings.ELASTICSEARCH_CARDINALITY_PRECISION:
```

Comment:
This feels like a small thing, but when we do our estimated counts, are they guaranteed to be higher than the truth? In other words, let's say `PRECISION` is set to 2000, and there are actually 2001 results. is it possible that the estimated count could be 1999? 

If that's true, CL would confidently say, "1999 results" when it should say "About 1999 results", right? 

In our documentation somewhere, you had a comment, I think, that said how far off the cardinality count could be. I think it might have been 5%? 

So maybe this line becomes:

```suggestion
    # Estimates can be below reality by as much as 5%. Round numbers below that threshold.
    lowest_possible_estimate = settings.ELASTICSEARCH_CARDINALITY_PRECISION * 0.95
    if search_count > lowest_possible_estimate:
```

Of course, if this is true, we'll want to tweak this in a few other places too.

Reply:
Thanks for noticing this. Yeah the max error described in the ES documentation is [6%](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-cardinality-aggregation.html#_counts_are_approximate)  and I confirmed and it can be positive or negative (up to ±6%).

For instance, in this search, counts are lower in the API (using the cardinality aggregation):

![Screenshot 2024-08-01 at 9 30 35 a m](https://github.com/user-attachments/assets/50eec306-eacf-4c44-ae71-9b87563103a0)
![Screenshot 2024-08-01 at 9 29 59 a m](https://github.com/user-attachments/assets/6b317f16-2f1e-46e5-af8d-6801e7be07c8)

So, I've applied your suggestion to compute the `lowest_possible_estimate`, which, based on our current threshold of 2000, would be `1880`. This will be rounded to `1800`, meaning that in the frontend, when the search counts are equal to or above `1800`, we'll use the word "About".

![Screenshot 2024-08-01 at 10 26 20 a m](https://github.com/user-attachments/assets/419183e4-5202-40fe-97c5-a2bddd0f46ba)